src/routers/alpha-router/functions/get-candidate-pools.ts

```diff
@@ -58,44 +58,47 @@ import {
   WGLMR_MOONBEAM,
   WMATIC_POLYGON,
   WMATIC_POLYGON_MUMBAI,
-  WXDAI_GNOSIS,
   WSTETH_MAINNET,
+  WXDAI_GNOSIS,
 } from '../../../providers/token-provider';
-import {
-  IV2PoolProvider,
-  V2PoolAccessor,
-} from '../../../providers/v2/pool-provider';
-import {
-  IV3PoolProvider,
-  V3PoolAccessor,
-} from '../../../providers/v3/pool-provider';
-import {
-  IV3SubgraphProvider,
-  V3SubgraphPool,
-} from '../../../providers/v3/subgraph-provider';
+import { IV2PoolProvider, V2PoolAccessor, } from '../../../providers/v2/pool-provider';
+import { IV3PoolProvider, V3PoolAccessor, } from '../../../providers/v3/pool-provider';
+import { IV3SubgraphProvider, V3SubgraphPool, } from '../../../providers/v3/subgraph-provider';
 import { unparseFeeAmount, WRAPPED_NATIVE_CURRENCY } from '../../../util';
 import { parseFeeAmount } from '../../../util/amounts';
 import { log } from '../../../util/log';
 import { metric, MetricLoggerUnit } from '../../../util/metric';
 import { AlphaRouterConfig } from '../alpha-router';
 
 export type PoolId = { id: string };
-export type CandidatePoolsBySelectionCriteria = {
+export type CandidatePoolsBySelectionCriteria<SubgraphPool> = {
```

Comment:
I had to add this generic to be able to have access (via the typecheck) to the candidatePools of each protocol, it was just choosing `PoolId` before which hid all the other fields I needed for it